test/e2e/suite/trustpolicy/trustpolicy_test.go

```diff
@@ -0,0 +1,467 @@
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticTimestamp was set to \"log\"",
+					"error: certificate \"O=Internet Widgits Pty Ltd,ST=Some-State,C=AU\" is not valid anymore, it was expired",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test override strict signatureVerification level with Authenticity set to log", func() {
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("override_strict_trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2ELocalKeysDir, "new_e2e.crt")))
+			// the artifact signed with a different cert from the cert in
+			// trust store.
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticity was set to \"log\"",
+					"signature is not produced by a trusted signer",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test override permissive signatureVerification level with Expiry set to enforced", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("override_permissive_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-expired-signature", "")
+
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchErrKeyWords("expiry validation failed.",
+					VerifyFailed)
+		})
+	})
+
+	It("test override permissive signatureVerification level with Authentic timestamp set to enforced", func() {
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("override_permissive_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-with-expired-cert", "")
+
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2EConfigPath, "localkeys", "expired_e2e.crt")))
+
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchErrKeyWords("authenticTimestamp validation failed",
+					VerifyFailed)
+		})
+	})
+
+	It("test override permissive signatureVerification level with Authenticity set to log", func() {
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("override_permissive_trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2ELocalKeysDir, "new_e2e.crt")))
+
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticity was set to \"log\"",
+					"signature is not produced by a trusted signer",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test invalid trust store", func() {
```

Comment:
This example is testing whether a specified trust store exists or not. How about the trust store specified in the trust policy is malformed? For instance, `"trustStores": ["ca:"],`

Reply:
> `"trustStores": ["ca:"],` 

We have unit tests for this already

Reply:
Added.